### PR TITLE
add github actions

### PR DESCRIPTION
**Pull request by isabellaliu77**
Signed-off-by: isabellaliu <isabella77@gmail.com>

## Description
Add github action support

## How Has This Been Tested?
I have tested it in my own repo.

## How are existing users impacted? What migration steps/scripts do we need?

No code changed, only the github action.

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/inlets/inlets/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    - name: Set up Go 1.x
+      uses: actions/setup-go@v2
+      with:
+        go-version: ^1.13
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Get dependencies
```
Comment (alexellis) [inline .github/workflows/ci.yaml:23]: This step can be removed.

File: Makefile [hunk 2]
```diff
@@ -1,12 +1,22 @@
...
+	CGO_ENABLED=0 go test $(shell go list ./... | grep -v /vendor/|xargs echo) -cover
+
+.PHONY: gofmt
+gofmt:
+	@test -z $(shell gofmt -l -s $(SOURCE_DIRS) ./ | tee /dev/stderr) || (echo "[WARN] Fix formatting issues with 'make fmt'" && exit 1)
 
 .PHONY: dist
 dist:
+	mkdir -p bin/
+	rm -rf bin/inlet*
```
Comment (alexellis) [inline Makefile:19]: This should be `bin/inlets*`

File: .github/workflows/publish.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 1
+      - name: Make
+        run: make all
+      - name: Upload release binaries
+        uses: alexellis/upload-assets@0.1.21
```
Comment (alexellis) [inline .github/workflows/publish.yaml:18]: This version needs to be changed to latest please.

https://github.com/alexellis/upload-assets/releases/

**Comment by isabellaliu77** [comment]
Hi Alex:
I have done the changes, Any problem, please let me know.

**Review by alexellis — APPROVED** [review]
LGTM

**Comment by alexellis** [comment]
Thank you @isabellaliu77 this looks good.

We actually have many more projects to migrate too and are co-ordinating on OpenFaaS Slack, you are welcome to join the `#contributors` channel there.

https://github.com/openfaas/faas/issues/1585

https://slack.openfaas.io/

**Pull request merged**